bin/scripts/fix_broken_ratings.js

```diff
@@ -0,0 +1,91 @@
+                    countlyDb.collection('app_users' + app._id).findOne({did: fixed_device_id}, function(err, correctUser) {
+                        if (!err) {
+                            // if there is correct app_user for fixed_device_id, use that values, otherwise use brokenUser uid and fixed did
+                            var correctReference = (correctUser) ? correctUser : {
+                                did: fixed_device_id,
+                                uid: brokenUser.uid
+                            };
+                            // map correct uid and did values to drill events records
+                            countlyDrillDB.collection("drill_events" + crypto.createHash('sha1').update("[CLY]_star_rating" + app._id).digest('hex'))
+                                .updateMany({did: brokenUser.did}, {
+                                    $set: {
+                                        did: correctReference.did,
+                                        uid: correctReference.uid
+                                    }
+                                }, function(err) {
+                                    if (!err) {
+                                        console.log("Drill event fixed for " + app.name);
+                                        // map correct uid and did values to broken feedback records
+                                        countlyDb.collection("feedback" + app._id)
+                                            .updateMany({device_id: brokenUser.did}, {
+                                                $set: {
+                                                    device_id: correctReference.did,
+                                                    uid: correctReference.uid
+                                                }
+                                            }, function(err) {
+                                                if (!err) {
+                                                    console.log("Feedback fixed for " + app.name);
+                                                    // if there is correct user for this fixed device_id, remove broken app_user record.
+                                                    if (correctUser) {
+                                                        countlyDb.collection('app_users' + app._id).remove({_id: brokenUser._id}, function(err) {
+                                                            doneUser();
+                                                        });
+                                                    }
+                                                    // if there is no correct user for this fixed device_id, fix this broken app_user record and use it as user.
+                                                    else {
+                                                        countlyDb.collection('app_users' + app._id).findAndModify({_id: brokenUser._id}, {}, {$set: {did: fixed_device_id}}, function(err) {
```

Comment:
_id field of user document actually depends on device_id (https://github.com/Countly/countly-server/blob/master/api/utils/requestProcessor.js#L1109-L1111) so if device_id changes, then _id should also change.
And you can't change _id of the document in MongoDB, so you need to delete this user, and insert new one with same properties + corrected _id and did